src/api/customerInstitutionsApi.ts

```diff
@@ -1,17 +1,32 @@
-import mockCustomerInstitutions from '../utils/testfiles/mock_customer_institutions.json';
 import { CustomerInstitution } from '../types/customerInstitution.types';
 import { getIdToken } from './userApi';
 import Axios from 'axios';
 import i18n from '../translations/i18n';
 import { StatusCode } from '../utils/constants';
 
-export enum CustomerInstituionApiPaths {
+export enum CustomerInstitutionApiPaths {
   CUSTOMER_INSTITUTION = '/customer',
 }
 
 export const getAllCustomerInstitutions = async () => {
-  // TODO: get all  publications
-  return mockCustomerInstitutions;
+  try {
+    const idToken = await getIdToken();
+    const headers = {
+      Authorization: `Bearer ${idToken}`,
+    };
+    const response = await Axios.get(CustomerInstitutionApiPaths.CUSTOMER_INSTITUTION, {
+      headers,
+    });
+    if (response.status === StatusCode.OK) {
+      return response.data;
+    } else {
+      return null;
```

Comment:
kom borti akkurat dette ifb nye endepunkt for institutions nå: hva oppnår vi egentlig med å returnere null i disse tilfellene? det jeg sa tidligere i dag med `if (!response || response.error) { dispatch(setNotification(response.error))}` for konsumentene blir jo feil siden response kan være null :p 

Burde vi heller ha noe sånn?
```
try {
   (...)
   if (response.status === StatusCode.OK) {
      return response.data;
   }
} // catch {}

return {
     error: i18n.t('feedback:error.get_customers'),
 };
```

Reply:
sant det! det vi gjør nå er jo å kun sjekke på StatusCode.OK, og det kan jo være at vi får en annen statuskode enn 200. tenker du at vi bare dropper `else`-en og lar catch ta av seg alt? eller mente du sånn som du skrev over der? er det egentlig god praksis å returnere noe etter en try-catch?

Reply:
hvis vi dropper else uten return etter catch så blir det returnert `undefined` hvis response har status 202 da

Reply:
jeg fjerner `!response` fra `if (!response || response.error) { dispatch(setNotification(response.error))}` på alle GET-kall hvertfall. Men hvis denne `const idToken = await getIdToken();` feiler f.eks, så må vi jo fange det opp i catch-en?

Reply:
eller vil en tom catch gjøre at den "svelger" feilen, også blir returnen utafor returnere?

Reply:
vi må uansett bruke catch i fremtiden for logging tror jeg. for hvis ikke er det vanskelig å skjønne hva som gikk galt

Reply:
men ser vi ikke har noen standard på dette, nei. ganske mange forskjellige måter vi har gjort det på i api-filene

Reply:
med en tom catch går den bare videre ja.

```
export const getInstitutions = async () => {
  const url = `${InstituionApiPaths.INSTITUTIONS}`;
  try {
    const idToken = await getIdToken();
    const headers = {
      Authorization: `Bearer ${idToken}`,
    };
    const response = await Axios.get(url, { headers });
    if (response.status === StatusCode.OK) {
      return response.data;
    }
  } catch {}

  return i18n.t('feedback:error.get_institutions');
};
```

Men liker det egentlig ikke, siden det siste return egentlig skal være happypath imo

Reply:
> hvis vi dropper else uten return etter catch så blir det returnert `undefined` hvis response har status 202 da

men er det riktig at vi skal returnere feilmelding hvis det er statuskode 202 da?

Reply:
hehe, der har du skrivefeilen også: `InstituionApiPaths `

Reply:
vi vil vel fortsatt ha sjekken for `response.status === StatusCode.OK`? hvis status plutselig er 202 så vil jo ikke noe returneres uten en else med mindre vi har noe etter try/catch også

Reply:
nei, venta

Reply:
`try-catch` er jo laget for happypaths da, så blir rart å returnere data på slutten ved bruk av det, men skjønner hva du mener ja.

Reply:
tror faktisk jeg helst vil returnere error for alt som ikke har statusen vi forventer ja. Kan jo være 404 eller 500 også. så får vi håndtere flere statuser der det trengs

Reply:
ja, tanken var vel etterhvert å ha en switch-case på statuskoder for alle gyldige koder, og slik at vi kan gi mer spesifikke errormeldinger til brukeren også

Reply:
mistenker vi kommer til å få mange bugs hvis vi bytter ut `else return null` mange plasser. men vi kan jo ta det litt etter litt da. Da endrer jeg til å bruke tom catch, med return etterpå, også kan vi være litt obs på det neste gang vi knytter opp mot API-endepunkt

Reply:
det gir mening ja. hvar tror du om å gjøre det sånn enn så lenge:
```
  try {
    const idToken = await getIdToken();
    const headers = {
      Authorization: `Bearer ${idToken}`,
    };
    const response = await Axios.get(url, { headers });
    if (response.status === StatusCode.OK) {
      return response.data;
    } else {
      return i18n.t('feedback:error.get_institutions');
    }
  } catch {
    return i18n.t('feedback:error.get_institutions');
  }
```

Reply:
ja, jeg er mer komfortabel med det enn en tom catch egentlig. synes det er rart å bruke try-catch, hvis vi ikke skal bruke catch'en

Reply:
nice, case closed! :D